base_multi_store/models/res_users.py

```diff
@@ -43,3 +44,39 @@ def __init__(self, pool, cr):
         # duplicate list to avoid modifying the original reference
         self.SELF_READABLE_FIELDS = list(self.SELF_READABLE_FIELDS)
         self.SELF_READABLE_FIELDS.append('store_id')
+
+    @api.constrains('store_id')
+    def _check_store_id(self):
+        for rec in self:
+            if rec.store_id and rec.store_id not in rec.store_ids:
+                raise ValidationError(
+                    _("The selected store it's not allow for your user"))
+
+    @api.model
+    def create(self, values):
+        values = self._remove_reified_groups(values)
+        user = super().create(values)
+        group_multi_store = self.env.ref(
+            'base_multi_store.group_multi_store', False)
+        if group_multi_store and 'store_ids' in values:
+            if len(user.store_ids) <= 1 and user.id in group_multi_store.users.ids:
+                user.write({'groups_id': [(3, group_multi_store.id)]})
+            elif len(user.store_ids) > 1 and user.id not in group_multi_store.users.ids:
+                user.write({'groups_id': [(4, group_multi_store.id)]})
+        return user
+
+    def write(self, values):
```

Comment:
are you duplicating `write` method?

Reply:
Thanks for the observation, i already fixed them now.